sources/Security.php

```diff
@@ -840,40 +822,48 @@ function createToken($action, $type = 'post')
  *
  * @param string $action
  * @param string $type = 'post' (get, request, or post)
- * @param bool $reset = true
+ * @param bool $reset = true Reset the token on failure
  * @param bool $fatal if true a fatal_lang_error is issued for invalid tokens, otherwise false is returned
  * @return boolean except for $action == 'login' where the token is returned
  */
 function validateToken($action, $type = 'post', $reset = true, $fatal = true)
 {
-	$type = $type == 'get' || $type == 'request' ? $type : 'post';
+	$type = ($type === 'get' || $type === 'request') ? $type : 'post';
 	$token_index = $type . '-' . $action;
 
-	// Logins are special: the token is used to has the password with javascript before POST it
+	// Logins are special: the token is used to have the password with javascript before POST it
```

Comment:
Could it be the original intent of the comment was "... is used to hash the password ..."?